lib/dialects/db2/connection-manager.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+const AbstractConnectionManager = require('../abstract/connection-manager');
+const Promise = require('../../promise');
+const { logger } = require('../../utils/logger');
+const sequelizeErrors = require('../../errors');
+const DataTypes = require('../../data-types').db2;
+const parserStore = require('../parserStore')('db2');
+const debug = logger.debugContext('connection:db2');
+
+class ConnectionManager extends AbstractConnectionManager {
+  constructor(dialect, sequelize) {
+    super(dialect, sequelize);
+    this.lib = this._loadDialectModule('ibm_db');
+    this.refreshTypeParser(DataTypes);
+  }
+
+  _refreshTypeParser(dataType) {
+    parserStore.refresh(dataType);
+  }
+
+  _clearTypeParser() {
+    parserStore.clear();
+  }
+
+  getdb2Config(config) {
+    const connObj = {
+      database: config.database,
+      hostname: config.host,
+      port: config.port,
+      uid: config.username,
+      pwd: config.password
+    };
+
+    if (config.ssl) {
+      connObj['security'] = config.ssl;
+    }
+    if (config.sslcertificate) {
+      connObj['SSLServerCertificate'] = config.sslcertificate;
+    }
+    if (config.dialectOptions) {
+      for (const key of Object.keys(config.dialectOptions)) {
+        connObj[key] = config.dialectOptions[key];
+      }
+    }
+    return connObj;
+  }
+
+  connect(config) {
+    return new Promise((resolve, reject) => {
+      const connectionConfig = this.getdb2Config(config);
+      const connection = new this.lib.Database();
+      connection.lib = this.lib;
+      connection.open(connectionConfig, err => {
+        if (err) {
+          let connError;
+          if (err.message && err.message.search('SQL30081N') != -1) {
+            connError = new sequelizeErrors.ConnectionRefusedError(err);
+          } else {
+            connError = new sequelizeErrors.ConnectionError(err);
+          }
+          connError['message'] = err.message;
+          return reject(connError);
+        }
+        debug('connection acquired');
+        return resolve(connection);
+      });
+    });
+  }
+
+  disconnect(connection) {
+    // Don't disconnect a connection that is already disconnected
+    if (connection.connected) {
+      connection.close(err => {
```

Comment:
Rename this to `error` please

Reply:
Done in this [commit](https://github.com/ibmdb/sequelize/commit/df7fae1086f11bd75e1e94e77c05799cb0f9542a). Thanks.